### PR TITLE
Use constructor injection and update null annotations

### PR DESCRIPTION
**Pull request by wborn**
There's also https://github.com/openhab/openhab-webui/pull/258 to update the Web UIs for these changes.

### CODE REVIEW

File: bundles/org.openhab.core.ui.icon/src/test/java/org/openhab/core/ui/icon/internal/IconServletTest.java [hunk 1]
```diff
@@ -74,6 +76,9 @@ public void reset() {
     private IconServlet servlet;
     private ByteArrayServletOutputStream responseOutputStream = new ByteArrayServletOutputStream();
 
+    private @Mock HttpContext httpContext;
```
Comment (cweitkamp) [inline bundles/org.openhab.core.ui.icon/src/test/java/org/openhab/core/ui/icon/internal/IconServletTest.java:79]: Don't you want to call them mock here?

```suggestion
    private @Mock HttpContext httpContextMock;
```

File: bundles/org.openhab.core.config.discovery/src/main/java/org/openhab/core/config/discovery/internal/DiscoveryServiceRegistryImpl.java [hunk 2]
```diff
@@ -136,8 +135,6 @@ public void reduceNumberOfDiscoveryServices() {
 
     private final Logger logger = LoggerFactory.getLogger(DiscoveryServiceRegistryImpl.class);
 
-    private @NonNullByDefault({}) SafeCaller safeCaller;
```
Comment (cweitkamp) [inline bundles/org.openhab.core.config.discovery/src/main/java/org/openhab/core/config/discovery/internal/DiscoveryServiceRegistryImpl.java:139]: :+1:

File: itests/org.openhab.core.config.discovery.usbserial.linuxsysfs.tests/src/main/java/org/openhab/core/config/discovery/usbserial/linuxsysfs/internal/PollingUsbSerialScannerTest.java [hunk 3]
```diff
@@ -37,22 +39,20 @@
 
     private UsbSerialDeviceInformationGenerator usbDeviceInfoGenerator = new UsbSerialDeviceInformationGenerator();
 
-    private UsbSerialScanner usbSerialScanner;
     private PollingUsbSerialScanner pollingScanner;
-    private UsbSerialDiscoveryListener discoveryListener;
+    private @Mock UsbSerialDiscoveryListener discoveryListenerMock;
+    private @Mock UsbSerialScanner usbSerialScanner;
```
Comment (cweitkamp) [inline itests/org.openhab.core.config.discovery.usbserial.linuxsysfs.tests/src/main/java/org/openhab/core/config/discovery/usbserial/linuxsysfs/internal/PollingUsbSerialScannerTest.java:44]:
```suggestion
    private @Mock UsbSerialScanner usbSerialScannerMock;
```

File: bundles/org.openhab.core.ui.icon/src/test/java/org/openhab/core/ui/icon/internal/IconServletTest.java (hunk 1 above)
Comment (wborn) [reply bundles/org.openhab.core.ui.icon/src/test/java/org/openhab/core/ui/icon/internal/IconServletTest.java:79]: Good catch, I'll update them!

**Pull request merged**